### PR TITLE
Fix issue when `transform` keyword passed to `GenericMap.draw_contours()`

### LINKED ISSUE #7191: `GenericMap.draw_contours()`: don't run internal transform code if `transform` keyword is provided
#5398 fixed some issues in `GenericMap.draw_contours()` by internally transforming the map to the axes WCS if the map WCS was different.  However, we should run this code only if the `transform` keyword argument was not provided, because we shouldn't do the internal transformation if the user is providing a specific transformation to use.

### PR DESCRIPTION
**Pull request by hayesla**
Fixes #7191 

I think this now address the issue outlined

### CODE REVIEW

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -2350,7 +2350,12 @@ def draw_contours(self, levels, axes=None, **contour_args):
         # We do this instead of using the `transform` keyword argument so that Matplotlib does not
         # get confused about the bounds of the contours
         if self.wcs is not axes.wcs:
-            transform = axes.get_transform(self.wcs) - axes.transData  # pixel->pixel transform
+            if "transform" in contour_args:
+                transform_orig = contour_args["transform"]
+                contour_args.pop("transform")
```
Comment (ayshih) [inline sunpy/map/mapbase.py:2355]:
```suggestion
                transform_orig = contour_args.pop("transform")
```

**Review by ayshih — COMMENTED** [review]
Needs a unit test, of course, which will presumably have to be a figure test

**Review by dstansby — CHANGES_REQUESTED** [review]
Looks like this still needs a test adding.

**Comment by nabobalis** [comment]
Superseded by https://github.com/sunpy/sunpy/pull/7427

**Pull request closed without merging**